src/main/java/org/team1540/robot2023/commands/vision/DriveToGamePiece.java

```diff
@@ -0,0 +1,34 @@
+package org.team1540.robot2023.commands.vision;
+
+import com.pathplanner.lib.PathPoint;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import java.util.function.DoubleSupplier;
+import org.team1540.robot2023.commands.drivetrain.Drivetrain;
+
+public class DriveToGamePiece extends CommandBase{
+    Drivetrain drivetrain;     
+    DoubleSupplier hypotenuse; 
+
+    public DriveToGamePiece(Drivetrain drivetrain, DoubleSupplier hypotenus){
+        this.drivetrain = drivetrain; 
+        this.hypotenuse = hypotenus; 
+    }
+    
+     /**
+     * @param Hypotenuse Hypotenuse to game piece in meters 
+     * @return PathPoint of gamepiece 
+     */
+    public PathPoint calculateLength() {
+       double length = Math.sin(drivetrain.getYaw().getDegrees()) * hypotenuse.getAsDouble(); 
+       double width = Math.cos(drivetrain.getYaw().getDegrees()) * hypotenuse.getAsDouble(); 
```

Comment:
This is probably a non-issue, but you should be storing the output of the supplier as a variable, not calling it twice. There's no assurance that it will give you the same thing both times.